packages/devtools_app/lib/src/shared/preferences.dart

```diff
@@ -82,8 +103,84 @@ class InspectorPreferencesController {
         _hoverEvalMode.value.toString(),
       );
     });
+  }
 
-    setGlobal(InspectorPreferencesController, this);
+  void _initCustomPubRootListeners() {
+    _customPubRootDirectories.addListener(() {
+      storage.setValue(
+        _customPubRootDirectoriesStorageId,
+        jsonEncode(_customPubRootDirectories.value),
+      );
+    });
+    _busyCounter.addListener(() {
```

Comment:
these listeners will need to be disposed. Look at the AutoDisposeControllerMixin